services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -157,12 +158,14 @@ private boolean isEfgsEnforceDateBasedDownloadAndNotSeen(String batchTag) {
     LocalDate date = batchInfo.getDate();
     String batchTag = batchInfo.getBatchTag();
     logger.info("Processing batch for date {} and batchTag {}.", date, batchTag);
+    AtomicReference<Optional<String>> nextBatchTag = new AtomicReference<>(Optional.empty());
     try {
       BatchDownloadResponse response = federationGatewayDownloadService.downloadBatch(batchTag, date);
+      AtomicBoolean batchContainsInvalidKeys = new AtomicBoolean(false);
       if (config.isBatchAuditEnabled()) {
         federationGatewayDownloadService.auditBatch(batchTag, date);
       }
-      AtomicBoolean batchContainsInvalidKeys = new AtomicBoolean(false);
+      nextBatchTag.set(response.getNextBatchTag());
```

Comment:
I guess this have to move before the audit happens

Reply:
This has become obsolete, since the audit call was moved down with another PR